report/src/report.cpp

```diff
@@ -67,7 +73,7 @@ struct ExtLogConfig : public scp::LogConfig {
     auto match(const char* type) -> bool { return regex_search(type, reg_ex); }
 };
 
-thread_local ExtLogConfig log_cfg;
+ExtLogConfig log_cfg;
```

Comment:
By removing the thread_local here the whole thing is not thread safe anymore since the (re-)init_logging functions modify it without a guard.

Reply:
But leaving it as thread_local makes the whole thing not work in a multi-threaded environment. So what you're saying is that we should put a guard in init, in case 2 threads call the init?.

Reply:
(Actually that lock already exists within configure_logging - maybe it could be pulled out, but if you have two things 'fighting' over what sort of logging you want, and they have different setups, you're going to get non-deterministic behaviour. I dont think a lock is actually going to help much????)

Reply:
Actually the orignal implementation (with thread_local) is thread save and used in such an environment where 2 SystemC kernels run in 2 threads each having their own sc_main with the initialization of the logging infrastructure. The only downside is that you do not have a ExtLogConfig object where all threads share the instance. This boils down to one kernel with many threads vs. many kernels with one thread each.

But to make it thread save all the (re-)init_logging(..) and set_logging_level(..) would need to lock a mutex to protect the log_cfg instance

Reply:
> Actually the orignal implementation (with thread_local) is thread save and used in such an environment where 2 SystemC kernels run in 2 threads each having their own sc_main with the initialization of the logging infrastructure. The only downside is that you do not have a ExtLogConfig object where all threads share the instance. This boils down to one kernel with many threads vs. many kernels with one thread each.
> 
> But to make it thread save all the (re-)init_logging(..) and set_logging_level(..) would need to lock a mutex to protect the log_cfg instance

Ahh, I see - yes - maybe this should be an option - it's kind of a choice about the use case.
a #def ?

Reply:
Fine with me...

Reply:
> I was thinking about it. How would you name it?

How about DISABLE_REPORT_THREAD_LOCAL

Reply:
I'll roll something

Reply:
how does that look?